app/util/fs.py

```diff
@@ -131,9 +132,17 @@ def zip_directory(target_dir: str, archive_filename: str) -> str:
     """
     # Create the archive in a temp location and then move it to the target dir.
     # (Otherwise the resulting archive will include an extra zero-byte file.)
-    tmp_path = shutil.make_archive(tempfile.mktemp(), 'zip', target_dir)
     target_path = os.path.join(target_dir, archive_filename)
-    shutil.move(tmp_path, target_path)
+    with tempfile.TemporaryDirectory() as temp_dirpath:
```

Comment:
@gcurtis Yeah, definitely. I thought about doing that but went this route instead just because it felt slightly more awkward to create a NamedTemporaryFile, immediately close it, and then reopen it with zipfile.